src/test/java/homework4/CollectionsIntoMapTest.java

```diff
@@ -0,0 +1,45 @@
+package homework4;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import static homework4.CollectionsIntoMap.convertCollectionsIntoMap;
+import static homework4.CollectionsIntoMap.convertCollectionsIntoMapException;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class CollectionsIntoMapTest {
+
+    ArrayList<Integer> firstFiveDimensionalList = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
+    ArrayList<Integer> secondFiveDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9, 10));
+    ArrayList<Integer> secondFourDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9));
+    ArrayList<Integer> secondSixDimensionalList = new ArrayList<>(Arrays.asList(6, 7, 8, 9, 10, 11));
+
+    @Test
+    void twoCollectionsSameSize() {
+        assertEquals(Map.of(1, 6, 2, 7, 3, 8, 4, 9, 5, 10), convertCollectionsIntoMap(firstFiveDimensionalList, secondFiveDimensionalList));
+    }
+
+    @Test
+    void firstCollectionSizeGreaterThanSecond() {
+        assertEquals(Map.of(1, 6, 2, 7, 3, 8, 4, 9, 5, "null"), convertCollectionsIntoMap(firstFiveDimensionalList, secondFourDimensionalList));
+    }
+
+    @Test
+    void secondCollectionSizeGreaterThanFirst() {
+        assertEquals(Map.of(1, 6, 2, 7, 3, 8, 4, 9, 5, 10), convertCollectionsIntoMap(firstFiveDimensionalList, secondSixDimensionalList));
+    }
+
+    @Test
+    void firstCollectionSizeGreaterThanSecondWithException() throws Exception {
```

Comment:
Does this current implementation work for you? I have the compilation error:
<img width="696" alt="image" src="https://user-images.githubusercontent.com/90613721/182465953-f5a49176-c304-4b73-9a66-1a4e363c9493.png">

Exceptions can be verified the next way:
<img width="1101" alt="image" src="https://user-images.githubusercontent.com/90613721/182465841-bbf1ce74-adba-43ce-96ad-f59da84cc535.png">

Reply:
Thanks, done